### DISCUSSION TITLE
SDXL super slow on M1 mac?

### DISCUSSION THREAD

**Discussion by ballenvironment** [huggingface/diffusers#7004] [Q&A]
SD1.5 runs pretty well on M1 mac getting around 2s/it. I tried switching to SDXL but it is massively slower, taking 40s/it and nearly 15 minutes to do a single image, and it looks worse than SD1.5.

Is this normal? The time is one thing but I don't know why the results are a lot worse than SD1.5 doesn't look like the other results people get using nvidia GPU

**Comment by sayakpaul** [comment]
Cc: @pcuenca

**Comment by pcuenca** [comment]
Hi @ballenvironment! SDXL requires more memory than SD1.5, because images are 4 times larger and there's more conditioning data too. Could it be the case that your computer does not have enough RAM for a pass through the model and is swapping to disk?

In my case (M1 Max 64 GB), running 25 steps through the base SDXL pipeline takes less than a minute, but memory use goes from about 24 GB to 38 GB.

**Reply by ballenvironment** [reply]
ah, I only have 16GB of ram. Does it only perform well with 64GB? That's unfortunate, I didn't think it was that much more intensive
